casanovo/denovo/model_runner.py

```diff
@@ -255,7 +262,36 @@ def initialize_trainer(self, train: bool) -> None:
                 strategy=self._get_strategy(),
                 val_check_interval=self.config.val_check_interval,
                 check_val_every_n_epoch=None,
+                log_every_n_steps=1,
             )
+
+            if self.config._user_config.get("log_metrics"):
+                if not self.output_dir:
+                    logger.warning(
+                        "Output directory not set in model runner. "
+                        "No loss file will be created."
+                    )
+                else:
+                    CSV_LOG_SUB_DIR = "csv_logs"
+                    if self.overwrite_ckpt_check:
+                        utils.check_dir_file_exists(
+                            self.output_dir,
+                            CSV_LOG_SUB_DIR,
+                        )
+
+                    additional_cfg.update(
+                        {
+                            "logger": lightning.pytorch.loggers.CSVLogger(
+                                self.output_dir,
+                                version=CSV_LOG_SUB_DIR,
```

Comment:
**question:** Shouldn't `name` be set instead of `version`?

Reply:
If you don't define a version PyLightning will automatically create version sub directories under `save_dir`. I tried setting `version` to an empty string and it still created the version sub directories (if I had to guess they have something like `if not version:` somewhere in their code). Since, in their own words, logs will be saved to `os.path.join(save_dir, name, version)` (https://lightning.ai/docs/pytorch/stable/extensions/generated/lightning.pytorch.loggers.CSVLogger.html), this is a work-around I came up with to save the logs directly to `<output_dir>/csv_logs`.